plugins/modules/ome_alert_policies_info.py

```diff
@@ -0,0 +1,193 @@
+        "Enabled": true,
+        "DefaultPolicy": true,
+        "PolicyData": {
+            "Catalogs": [],
+            "Severities": [
+                16
+            ],
+            "MessageIds": [],
+            "Devices": [],
+            "DeviceTypes": [],
+            "Groups": [],
+            "AllTargets": false,
+            "Schedule": {
+                "StartTime": null,
+                "EndTime": null,
+                "CronString": null,
+                "Interval": false
+            },
+            "Actions": [
+                {
+                    "Id": 5,
+                    "Name": "Mobile",
+                    "ParameterDetails": [],
+                    "TemplateId": 112
+                }
+            ],
+            "UndiscoveredTargets": []
+        },
+        "State": true,
+        "Visible": true,
+        "Owner": null,
+    }
+]
+error_info:
+  description: Details of the HTTP Error.
+  returned: on HTTP error
+  type: dict
+  sample: {
+    "error": {
+      "code": "Base.1.0.GeneralError",
+      "message": "A general error has occurred. See ExtendedInfo for more information.",
+      "@Message.ExtendedInfo": [
+        {
+          "MessageId": "GEN1234",
+          "RelatedProperties": [],
+          "Message": "Unable to process the request because an error occurred.",
+          "MessageArgs": [],
+          "Severity": "Critical",
+          "Resolution": "Retry the operation. If the issue persists, contact your system administrator."
+        }
+      ]
+    }
+  }
+'''
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.ome import RestOME, ome_auth_params
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import remove_key
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+
+
+ALERT_POLICY_URI = "AlertService/AlertPolicies"
+
+
+class OMEAlertPolicyInfo():
+
+    def __init__(self) -> None:
+        self.module = get_module_parameters()
+
+    def get_all_alert_policy_info(self, rest_obj) -> list:
+        resp = rest_obj.invoke_request("GET", ALERT_POLICY_URI)
+        value = resp.json_data["value"]
+        return remove_key(value)
+
+    def get_alert_policy_info(self, rest_obj) -> list:
+        result = []
+        policy_name = self.module.params.get("policy_name")
+        if policy_name is not None:
+            if policy_name == "":
+                return []
+            value = self.get_all_alert_policy_info(rest_obj)
+            for each_element in value:
+                if each_element["Name"] == policy_name:
+                    result = [each_element]
+                    break
+        else:
+            result = self.get_all_alert_policy_info(rest_obj)
```

Comment:
Instead of calling "self.get_all_alert_policy_info(rest_obj)" twice , cant we call it once and then use the same output if policy name is given. 
As output of this function is needed in both the cases whether policy name is given or we have to result all policy info.

Reply:
I'm not calling that function twice, both are under if ... else statement, only once it'll get trigger, not twice.

Reply:
I mean we can keep that function call outside if else coz in both the cases that will be called.